test/test_habitat_env.py

```diff
@@ -290,6 +291,74 @@ def test_rl_vectorized_envs(gpu2gpu):
                 ), "dones should be true after max_episode steps"
 
 
+@pytest.mark.parametrize("gpu2gpu", [False])
+def test_rl_vectorized_envs_batch_renderer(gpu2gpu):
+    import habitat_sim
+
+    if gpu2gpu and not habitat_sim.cuda_enabled:
+        pytest.skip("GPU-GPU requires CUDA")
+
+    configs, datasets = _load_test_data()
+    for config in configs:
+        with habitat.config.read_write(config):
+            config.habitat.simulator.enable_batch_renderer = True
+            config.habitat.simulator.habitat_sim_v0.enable_gfx_replay_save = (
+                True
+            )
+            config.habitat.simulator.create_renderer = False
+            config.habitat.simulator.habitat_sim_v0.gpu_gpu = gpu2gpu
+            agent_config = get_agent_config(config.habitat.simulator)
+            # Only keep the rgb_sensor
+            agent_config.sim_sensors = {
+                "rgb_sensor": agent_config.sim_sensors["rgb_sensor"]
+            }
+
+    num_envs = len(configs)
+    env_fn_args = tuple(zip(configs, datasets, range(num_envs)))
+    with habitat.VectorEnv(
+        make_env_fn=_make_dummy_env_func, env_fn_args=env_fn_args
+    ) as envs:
+        envs.initialize_batch_renderer(configs[0])
+        observations = envs.reset()
+        envs.post_step(observations)
```

Comment:
It looks like my old draft PR did the same thing, and I was calling `post_step` in `ppo_trainer.py`:
https://github.com/facebookresearch/habitat-lab/pull/863/files#diff-2e3ecd82d2bf278a7c6b6b6e50f6e189c2a7d35ce1e947c97b6d8bf82cea3b8b

Is this planned for a follow-up PR? We can't do any training with this until calls to `post_step` are integrated, right?

Reply:
> Ah, ok, so you expect the owner of VectorEnv (the calling code) to remember to call post_step in all the right places?

Yes. An alternative would be to automatically call `post_step` after `reset` or `step`, but the user needs to remember manually calling `post_step` if the `reset_at` or `step_at` variants are used. I'm not a fan of this because we need to unwrap env outputs to get observations, then pack them again after post processing.

I opted to keep the call separated, and refactor later if we think that this is a better idea.

> Is this planned for a follow-up PR? We can't do any training with this until calls to post_step are integrated, right?

Yes, I'm planning to add integration with rearrange sim and `ppo_trainer.py` immediately after.

Reply:
>if the reset_at or step_at variants are used
I don't think these are widely used. We should print a warning or fatal error if folks try to use these with the batch renderer.

@vincentpierre @mathfac what do you think about marking `reset_at` and `step_at` as deprecated? These are not batch-friendly and we should discourage their use.

Reply:
Also, I'm okay with requiring user code to call `post_step` explicitly.

Reply:
Also, I'm okay with requiring user code to call `post_step` explicitly.